src/project.ts

```diff
@@ -167,4 +167,84 @@ async function addEditorSettingsButton () {
 	}
 }
 
-export { addProgramInfo, keyboardShortcuts, addEditorSettingsButton, checkHiddenOrDeleted };
+function saveNewScratchpadToLocalStorage () {
+	let aceEditor;
+	let checkIfEditorIsReadyInterval = setInterval(function () {
+		aceEditor = document.getElementsByClassName("scratchpad-ace-editor")[0]?.env?.editor;
+
+		if (aceEditor !== undefined) {
+			// get the type of the program
+			const programType = window.location.href.split("/")[5];
+			if (!["pjs", "webpage", "sql"].includes(programType)) {
+				clearInterval(checkIfEditorIsReadyInterval); // we can stop checking if the editor is ready
+				return;
+			}
+
+			// get user kaid
+			let userKaid;
+			for (const key in window.__APOLLO_CLIENT__.cache.data.data) {
+				if (key.startsWith("User:kaid_")) {
+					userKaid = key.slice("User:".length, key.indexOf("."));
+				}
+			}
+			
+			const localStorageKey = "ka:4:cs-scratchpad-new" + userKaid + "-" + programType;
```

Comment:
I know I said earlier that it might be best to leverage KA's existing localStorage saves, but I didn't realize how unreliable those were.  At this point, I think it might be best make this key something like `kaeScrachpadSave-programtype` in order to avoid any future naming collisions.